src/main/java/com/gtnewhorizon/structurelib/ClientProxy.java

```diff
@@ -374,7 +379,7 @@ public void draw(Tessellator tes, double eyeX, double eyeY, double eyeZ, int eye
                             tes.addVertexWithUV(X, Y + size, Z, u, v);
                             break;
                         case 2:
-                            if ((Z >= eyeZ) != (j == 1)) continue;
+                            if ((worldZ >= eyeZ) != (j == 1)) continue;
```

Comment:
It is quite unfortunate that this code still uses ordinal face numbers as this leads to such unreadable unmaintainable logic